### ISSUE TITLE
`capa-iam-operator` CrashLooping on `grizzly` when attempting to handle EKS clusters

### ISSUE THREAD

**Issue by AverageMarcus** [giantswarm/roadmap#2932]
From the logs:

```
I1031 08:04:12.985396       1 controller.go:117]  "msg"="Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference" "AWSManagedControlPlane"={"name":"t-1q7mdyx","namespace":"org-t-2rwza89"} "controller"="awsmanagedcontrolplane" "controllerGroup"="controlplane.cluster.x-k8s.io" "controllerKind"="AWSManagedControlPlane" "name"="t-1q7mdyx" "namespace"="org-t-2rwza89" "reconcileID"="138dd5aa-8e59-4e37-9ece-573b9272952a"
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17e23f9]
```

Note that the EKS workload cluster has now been removed so `capa-iam-operator` is functioning correctly for now.

**Comment by calvix** [comment]
```
│ goroutine 457 [running]:                                                                                                                                           │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()                                                                             │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:118 +0x1f4                                                            │
│ panic({0x19a7180, 0x2e25d30})                                                                                                                                      │
│     /usr/local/go/src/runtime/panic.go:884 +0x213                                                                                                                  │
│ github.com/giantswarm/capa-iam-operator/pkg/iam.(*IAMService).GetIRSAOpenIDForEKS(0xc00061fcb0, {0xc0008e3c56, 0x9})                                               │
│     /workspace/pkg/iam/iam.go:621 +0xb9                                                                                                                            │
│ github.com/giantswarm/capa-iam-operator/controllers.(*AWSManagedControlPlaneReconciler).Reconcile(0xc00077ce00, {0x2048f00, 0xc0005b1800}, {{{0xc0008e3c60?, 0x0?} │
│     /workspace/controllers/awsmanagedcontrolplane_controller.go:145 +0xcf6                                                                                         │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x2048e58?, {0x2048f00?, 0xc0005b1800?}, {{{0xc0008e3c60?, 0x1b10e00?}, {0xc0008e3c │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:121 +0xc8                                                             │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0003dbf40, {0x2048e58, 0xc000516a50}, {0x1a16f60?, 0xc0003ac5a0?})        │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:320 +0x309                                                            │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0003dbf40, {0x2048e58, 0xc000516a50})                                  │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:273 +0x1d9                                                            │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()                                                                               │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:234 +0x85                                                             │
│ created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2                                                                        │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:230 +0x587                                                            │
│ Stream closed EOF for giantswarm/capa-iam-operator-67dcdfdcd7-pn6md (capa-iam-operator)                                                                            │
│           
```
https://github.com/giantswarm/capa-iam-operator/blob/947c35060267391c4b4d0a63c8325cba6947f788/pkg/iam/iam.go#L621

**Comment by calvix** [comment]
fixed
